pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/pkg/errors"
 )
 
-func (cs *ContainerService) setCustomCloudProfileDefaults() error {
+func (cs *ContainerService) setCustomCloudProfileDefaults(isUpgrade, isScale bool) error {
```

Comment:
Multiple parameters of the same time is a code smell. Please think about adding a struct as input so the fields will be easily differentiated.

Also, I feel like this we clean up else where recently.

https://dave.cheney.net/2019/09/24/be-wary-of-functions-which-take-several-parameters-of-the-same-type